utils/wrappers/rebalancingWrapper.ts

```diff
@@ -495,4 +498,131 @@ export class RebalancingWrapper {
+      auctionLibrary,
+      auctionTimeToPivot,
+      { from },
+    );
+
+    return new RebalancingTokenManagerContract(
+      new web3.eth.Contract(truffleRebalacingTokenManager.abi, truffleRebalacingTokenManager.address),
+      { from, gas: DEFAULT_GAS },
+    );
+  }
+
+  public getExpectedNextSetParameters(
+    btcPrice: BigNumber,
+    ethPrice: BigNumber,
+  ): any {
+    let units: BigNumber[];
+    let naturalUnit: BigNumber;
+    if (btcPrice.greaterThanOrEqualTo(ethPrice)) {
+      const ethUnits = btcPrice.mul(new BigNumber(10 ** 10)).div(ethPrice).round(0, 3);
+      units = [new BigNumber(1), ethUnits];
+      naturalUnit = new BigNumber(10 ** 10);
+    } else {
+      const btcUnits = ethPrice.mul(new BigNumber(100)).div(btcPrice).round(0, 3);
+      const ethUnits = new BigNumber(100).mul(new BigNumber(10 ** 10));
+      units = [btcUnits, ethUnits];
+      naturalUnit = new BigNumber(10 ** 12);
+    }
+
+    return {
+      units,
+      naturalUnit,
+    };
+  }
+
+  public async getExpectedAuctionParameters(
+    btcPrice: BigNumber,
+    ethPrice: BigNumber,
+    auctionTimeToPivot: BigNumber,
+    currentSetToken: SetTokenContract,
+  ): Promise<any> {
+    const THIRTY_MINUTES_IN_SECONDS = new BigNumber(30 * 60);
+
+    const nextSetParams = this.getExpectedNextSetParameters(
+      btcPrice,
+      ethPrice
+    );
+
+    const currentSetNaturalUnit = await currentSetToken.naturalUnit.callAsync();
+    const currentSetUnits = await currentSetToken.getUnits.callAsync();
+
+    const currentSetDollarAmount = this.computeTokenValue(
+      currentSetUnits,
+      currentSetNaturalUnit,
+      btcPrice,
+      ethPrice
+    );
+
+    const nextSetDollarAmount = this.computeTokenValue(
+      nextSetParams['units'],
+      nextSetParams['naturalUnit'],
+      btcPrice,
+      ethPrice
+    );
+
+    const fairValue = nextSetDollarAmount.div(currentSetDollarAmount).mul(1000).round(0, 3);
+    const onePercentSlippage = fairValue.div(100).round(0, 3);
+
+    const thirtyMinutePeriods = auctionTimeToPivot.div(THIRTY_MINUTES_IN_SECONDS).round(0, 3);
+    const halfPriceRange = thirtyMinutePeriods.mul(onePercentSlippage).div(2).round(0, 3);
+
+    const auctionStartPrice = fairValue.sub(halfPriceRange);
+    const auctionPivotPrice = fairValue.add(halfPriceRange);
+
+    return {
+      auctionStartPrice,
+      auctionPivotPrice,
+    };
+  }
+
+  public async getExpectedSetTokenAsync(
+    setTokenAddress: Address,
+  ): Promise<SetTokenContract> {
+    return new SetTokenContract(
+      new web3.eth.Contract(SetToken.abi, setTokenAddress),
+      { from: this._tokenOwnerAddress },
+    );
+  }
+
+  private computeTokenValue(
+    units: BigNumber[],
+    naturalUnit: BigNumber,
+    btcPrice: BigNumber,
+    ethPrice: BigNumber,
+  ): BigNumber {
+    const FULL_TOKEN_AMOUNT = new BigNumber(10 ** 18);
+    const WBTC_TOKEN_DECIMALS = new BigNumber(10 ** 8);
+    const WETH_TOKEN_DECIMALS = new BigNumber(10 ** 18);
```

Comment:
Same, all constants here. You should maybe add these to set-protocol-utils so they can be reused.